lib/autocomplete-manager.coffee

```diff
@@ -149,141 +150,190 @@ class AutocompleteManager
+
+    return prefix
+
+  # Private: Checks whether the current file is blacklisted.
+  #
+  # Returns {Boolean} that defines whether the current file is blacklisted
+  currentFileBlacklisted: ->
+    blacklist = (atom.config.get('autocomplete-plus.fileBlacklist') or '')
+      .split(',')
+      .map((s) -> s.trim())
 
-  onDidChangeItems: (cb) ->
-    @emitter.on 'did-change-items', cb
+    fileName = path.basename(@editor.getBuffer().getPath())
+    for blacklistGlob in blacklist
+      return true if minimatch(fileName, blacklistGlob)
 
-  # Private: Focuses the hidden input, starts listening to keyboard events
-  setActive: ->
-    @addKeyboardInteraction()
-    @active = true
+    return false
 
   # Private: Gets called when the content has been modified
   contentsModified: =>
-    delay = parseInt(atom.config.get "autocomplete-plus.autoActivationDelay")
-    if @delayTimeout
-      clearTimeout @delayTimeout
-
-    @delayTimeout = setTimeout @runAutocompletion, delay
+    delay = parseInt(atom.config.get('autocomplete-plus.autoActivationDelay'))
+    clearTimeout(@delayTimeout) if @delayTimeout
+    @delayTimeout = setTimeout(@runAutocompletion, delay)
 
   # Private: Gets called when the cursor has moved. Cancels the autocompletion if
-  # the text has not been changed and the autocompletion is
+  # the text has not been changed.
   #
   # data - An {Object} containing information on why the cursor has been moved
   cursorMoved: (data) =>
-    @cancel() unless data.textChanged
+    @hideSuggestionList() unless data.textChanged
 
   editorHasFocus: =>
+    return false unless @editorView?
     editorView = @editorView
     editorView = editorView[0] if editorView.jquery
     return editorView.hasFocus()
+
   # Private: Gets called when the user saves the document. Cancels the
-  # autocompletion
-  editorSaved: =>
+  # autocompletion.
+  bufferSaved: =>
     return unless @editorHasFocus()
-    @cancel()
+    @hideSuggestionList()
 
-  # Private: Cancels the autocompletion if the user entered more than one character
-  # with a single keystroke. (= pasting)
+  # Private: Cancels the autocompletion if the user entered more than one
+  # character with a single keystroke. (= pasting)
   #
   # e - The change {Event}
-  editorChanged: (e) =>
-    return if @compositionInProgress
+  bufferChanged: (e) =>
+    return if @suggestionList.compositionInProgress
     return unless @editorHasFocus()
-    if atom.config.get("autocomplete-plus.enableAutoActivation") and ( e.newText.trim().length is 1 or e.oldText.trim().length is 1 )
+    if atom.config.get('autocomplete-plus.enableAutoActivation') and (e.newText.trim().length is 1 or e.oldText.trim().length is 1)
       @contentsModified()
     else
-      @cancel()
-
-  # Private: Replaces the current prefix with the given match
+      @hideSuggestionList()
+
+  #  |||              |||
+  #  vvv PROVIDER API vvv
+
+  registerProviderForGrammars: (provider, grammars) =>
```

Comment:
> AutocompleteManager is to handle the editor and editor interactions.

Is it? I mean, I suppose it could be its sole purpose, but I think the Provider API makes sense there too, at least for now.